src/Data/Entity/Login.php

```diff
@@ -0,0 +1,56 @@
+<?php
+namespace App\Data\Entity;
+
+use App\Data\Authenticator;
+use App\Data\Entity;
+
+class Login extends Entity
+{
+
+    protected $transformer = 'App\Data\Transformer\LoginTransformer';
```

Comment:
Remember, if our `Entity` class extends `JsonSerializable` we get the advantage of automatic conversion when we use `json_encode`. So if we have:

```php
$objects = [
    new User(/* data */),
    new User(/* data */),
    new User(/* data */),
];
```

We can just do:
```php
$output['users'] = json_encode($objects);
```
To get:
```json
{
  "users": [
    {
      "id": 1
    },
    { ... },
    { ... }
  ]
}
```

I'm not sure about performance or anything, but it's really slick in my opinion. Not very flexible, but are we really going to use something besides JSON? :facepunch: